django/http/request.py

```diff
@@ -66,6 +68,12 @@ def __repr__(self):
             return '<%s>' % self.__class__.__name__
         return '<%s: %s %r>' % (self.__class__.__name__, self.method, self.get_full_path())
 
+    @property
```

Comment:
You can use `django.utils.functional.cached_property` here and remove the custom caching code.

Reply:
Nice! I missed that one. I remember seeing it but then I completely forgot.